libraries/botbuilder-dialogs/src/memory/scopes/settingsMemoryScope.ts

```diff
@@ -25,16 +35,110 @@ export class SettingsMemoryScope extends MemoryScope {
      * @param dc The [DialogContext](xref:botbuilder-dialogs.DialogContext) object for this turn.
      * @returns The memory for the scope.
      */
-    public getMemory(dc: DialogContext): object {
-        let settings: object = {};
+    public getMemory(dc: DialogContext): Record<string, unknown> {
+        let settings: Record<string, unknown> = {};
         if (dc.context.turnState.has(ScopePath.settings)) {
             settings = dc.context.turnState.get(ScopePath.settings);
-        }
-        for (const key in process.env) {
-            if (typeof process.env[key] == 'string') {
-                settings[key] = process.env[key];
+        } else {
+            const configuration = dc.context.turnState.get(DialogTurnStateConstants.configuration);
+            if (configuration) {
+                settings = SettingsMemoryScope.loadSettings(configuration);
+                dc.context.turnState.set(ScopePath.settings, settings);
             }
         }
+        const envSettings = SettingsMemoryScope.loadSettings(process.env);
+        merge(settings, envSettings);
         return settings;
     }
+
+    /**
+     * Build a dictionary view of configuration providers.
+     *
+     * @param configuration The configuration that we are running with.
+     * @returns Projected dictionary for settings.
+     */
+    protected static loadSettings(configuration: Record<string, string>): Record<string, unknown> {
+        const settings = {};
+        if (configuration) {
+            // load configuration into settings
+            const root = this.convertFlattenSettingToNode(Object.entries(configuration));
+            root.children.forEach((u) => {
+                settings[u.value] = this.convertNodeToObject(u);
+            });
+        }
+        return settings;
```

Comment:
I feel like this is the first time I'm seeing an implicit typing from the signature method on a created and assigned variable... Beautiful!